### DISCUSSION TITLE
[sdcard] How to delete/rename/move files? - Basic file operations

### DISCUSSION THREAD

**Discussion by BOND12** [gsantner/markor#1716] [Q&A] [answered]
I noticed that markor does not have basic file functionality when working with documents from the "Files" column. Such as delete, rename, move to folder, copy to folder. As a result, for example, if I make a mistake when entering the file name when creating it, I will have to open the documents folder using the file manager to fix it. It would be nice to add this functionality to markor itself.

**Comment by gsantner** [comment]
> . It would be nice to add this functionality to markor itself.

Ehm, Markor has a own file manager on the main screen? And it supports delete/rename/move/copy/...  
Long-press a file to select it. Then you can see buttons/icons in the top (blue menu). for copy/move/delete etc.

-----

"Files Colums" -> you mean you are at the editor?  
Correct yes, file management is at the Markor file manager. And the editor is for editing and viewing documents.

**Reply by BOND12** [reply]
I mean this window.

<a href="https://ibb.co/HXJ3JtR"><img src="https://i.ibb.co/HXJ3JtR/Screenshot-20220522-203840.png" alt="Screenshot-20220522-203840" border="0"></a>

I mean, this is the window where you can select files to view and edit. I didn't find any file functions in it, except for the create file function.

**Reply by gsantner** [reply]
Long-press a file to select it. Then you can see buttons/icons in the top (blue menu). for copy/move/delete etc.

**Reply by BOND12** [reply]
I don't see them for some reason. 
![Screenshot_20220522-210517](https://user-images.githubusercontent.com/56501627/169709412-49211227-f5dc-4bf8-a2ee-304181c5d1eb.png)
![Screenshot_20220522-210524](https://user-images.githubusercontent.com/56501627/169709415-2bdf9c83-7e0b-4adb-8758-6a2d59aecb44.png)

**Reply by gsantner** [reply]
Is this on a SD-Card? Then I see, for SD-Card we don't have some things implemented yet. But generally speaking these functions are available you can see them sure on internal storage.

**Reply by BOND12** [reply]
Okay, I understand that. Do you implement these functions for sd cards later?

**Reply by gsantner** [reply]
Well simply said, this is not a company. People implement what they are interested in and when they want to give their own time. But generally speaking yes we want to improve support for storage on Android.

**Comment by gsantner** [comment, marked as answer]
I made some improvements for SD-Card support recently. Please see <https://github.com/gsantner/markor/issues/603>.

**Reply by SugarRayLua** [reply]
Great app, thanks!
I just downloaded it.
However, I'm having trouble as well locating those basic file functions you mentioned.
I've attached a screenshot where I'm on the files tab of the app and selected a to do list and them the three dots in the top left but don't see any file buttons to delete or move.
Thanks 😊
![Screenshot_20260627-232413.png](https://github.com/user-attachments/assets/66123760-835f-4155-8621-8f5187b02eec)

Android 16
Markor 2.16.1 from F-Droid

**Reply by PhantomLasagne** [reply]
Hey mate!

I believe you can only perform file operations on files saved to your device file system, and not with Markor's To-Do & Quicknote items. Grain of salt as I haven't looked at the code and only recently started using Markor.

See below:

![ResizedImage_2026-07-24_12-28-48_9262.png](https://github.com/user-attachments/assets/9d436ab4-0139-40af-871f-2b458a899df4)
![ResizedImage_2026-07-24_12-22-39_6341[2].png](https://github.com/user-attachments/assets/196805fa-f721-4a5d-861c-a52064413aad)


![ResizedImage_2026-07-24_12-30-16_5240.png](https://github.com/user-attachments/assets/ecbf9864-d5dd-4ca8-9163-8f7bbb34c5e6)
![ResizedImage_2026-07-24_12-27-13_7088[0].png](https://github.com/user-attachments/assets/2632200f-12e6-4c8b-8cbe-9d06efac78bf)

**Comment by SugarRayLua** [comment]
Thanks, @PhantomLasagne 

I do now see how that works. If I used the + icon on the bottom right of the screen, created my own .md file, it then appears on the screen, and I can select it (which turns the .md file into a checkmark) then am able to move the file.

However, as you said, for some reason I can't actually move the included default "QuickNote.md" file. Not a big deal, but I was confused since on the main Github screen, it seems to say that the default QuickNote.md file is movable:

"What is QuickNote?
The fastest and easiest way to take notes! QuickNote is a file in Markdown format with a freely choosable file location. You can access it by swiping twice at the main screen, by selecting QuickNote at Notebook, or by using the dedicated launcher. The location of this file is freely choosable and independent from the Notebook directory."

**Reply by PhantomLasagne** [reply]
Yeah it had me confused for a while too @SugarRayLua 

Glad you got it sorted!

**Comment by gsantner** [comment]
You pick a different file in Markor settings. Not move the file. Which you can do, but not relates to the default loaded one.
